src/Agent.Worker/TaskRunner.cs

```diff
@@ -66,6 +129,31 @@ public async Task RunAsync()
             Definition definition = taskManager.Load(Task);
             ArgUtil.NotNull(definition, nameof(definition));
 
+            // Verify task signatures if a fingerprint is configured for the Agent.
+            var configurationStore = HostContext.GetService<IConfigurationStore>();
+            AgentSettings settings = configurationStore.GetSettings();
+            
+            if (!String.IsNullOrEmpty(settings.Fingerprint))
+            {
+                ISignatureService signatureService = HostContext.CreateService<ISignatureService>();
+                Boolean verificationSuccessful =  await signatureService.VerifyAsync(definition);
```

Comment:
Should we have some cancellation token with a timeout on the verification?

Reply:
Yup good call, I added the token from the ExecutionContext. I assume that's the right one.